src/Models/Type.php

```diff
@@ -134,15 +135,36 @@ public function getExpectedOrder(): array
         return $this->expectedOrder;
     }
 
-    /** @noinspection PhpUnhandledExceptionInspection */
-    public function isEnum($name): bool
+    public function isEnum(string $name): bool
     {
         $cast = $this->casts[$name] ?? null;
-        if ($cast) {
-            return (new ReflectionClass($cast))->isEnum();
+
+        if ($cast === null) {
+            return false;
+        }
+
+        // Έλεγχος αν το αποτέλεσμα είναι ήδη στην cache
+        if (isset($this->enumCache[$cast])) {
+            return $this->enumCache[$cast];
         }
 
-        return false;
+        // Έλεγχος αν η κλάση υπάρχει
+        if (!class_exists($cast)) {
+            $this->enumCache[$cast] = false;
+            return false;
+        }
+
+        try {
+            $reflection = new ReflectionClass($cast);
+            $isEnum = $reflection->isEnum();
+            // Αποθήκευση του αποτελέσματος στην cache
+            $this->enumCache[$cast] = $isEnum;
+            return $isEnum;
+        } catch (ReflectionException $e) {
```

Comment:
ReflectionException is not imported and $e is not used inside the catch block.

**Suggestion**
`use Throwable;`

Use `} catch (Throwable) {` instead and omit `$e` variable.

Reply:
On a second review it looks like the `ReflectionException` is never thrown because you already are checking if the class exists, so I think it's safe to remove the whole try catch block.